### PR TITLE
implement query to update an event trigger

### PR DESCRIPTION
**Pull request by tirumaraiselvan**

### CODE REVIEW

File: server/src-lib/Hasura/RQL/DDL/Subscribe.hs [hunk 1]
```diff
@@ -211,20 +230,65 @@ instance HDBQuery CreateEventTriggerQuery where
...
 
 instance HDBQuery DeleteEventTriggerQuery where
-  type Phase1Res DeleteEventTriggerQuery = ()
+  type Phase1Res DeleteEventTriggerQuery = QualifiedTable
   phaseOne = unsubTableP1
-  phaseTwo q _ = unsubTableP2 q
+  phaseTwo q qt = unsubTableP2 qt q
+  schemaCachePolicy = SCPReload
+
+upsubTableP1 :: (P1C m) => UpdateEventTriggerQuery -> m (QualifiedTable, EventTriggerDef)
```
Comment (0x777) [inline server/src-lib/Hasura/RQL/DDL/Subscribe.hs]: adminOnly

File: server/src-lib/Hasura/RQL/DDL/Subscribe.hs [hunk 2]
```diff
@@ -178,15 +198,21 @@ markForDelivery eid =
...
-subTableP1 :: (P1C m) => CreateEventTriggerQuery -> m (QualifiedTable, EventTriggerDef)
-subTableP1 (CreateEventTriggerQuery name qt insert update delete retryConf webhook) = do
+subTableP1 :: (P1C m) => CreateEventTriggerQuery -> m (QualifiedTable, Bool, EventTriggerDef)
+subTableP1 (CreateEventTriggerQuery name qt insert update delete retryConf webhook replace) = do
   adminOnly
   ti <- askTabInfo qt
+  -- can only replace for same table
+  when replace $ do
+    ti' <- askTabInfoFromTrigger name
+    when (ti' /= ti) $ throw400 NotSupported "cannot replace table or schema for trigger"
```
Comment (0x777) [inline server/src-lib/Hasura/RQL/DDL/Subscribe.hs]: just compare the tables?

File: server/src-lib/Hasura/RQL/Types/Subscribe.hs [hunk 3]
```diff
@@ -79,7 +79,8 @@ instance FromJSON CreateEventTriggerQuery where
     update    <- o .:? "update"
     delete    <- o .:? "delete"
     retryConf <- o .:? "retry_conf"
-    webhook   <- o .: "webhook"
+    webhook   <- o .:  "webhook"
+    mreplace   <- o .:? "replace"
```
Comment (0x777) [inline server/src-lib/Hasura/RQL/Types/Subscribe.hs]:
```haskell
 replace <- o .:? "replace" .!= False
```

**Comment by 0x777** [comment]
@praveenweb have the console parts been done?

**Comment by praveenweb** [comment]
@0x777 - console changes haven't been done. This can be released first for next version and then console can add the functionality for alpha19+ ?

**Comment by shahidhk** [comment]
issue for console update: https://github.com/hasura/graphql-engine/issues/489

**Pull request merged**